zebra/redistribute.c

```diff
@@ -531,14 +547,16 @@ void zebra_interface_address_add_update(struct interface *ifp,
 	struct listnode *node, *nnode;
 	struct zserv *client;
 	struct prefix *p;
+	char buf1[256];
 
 	if (IS_ZEBRA_DEBUG_EVENT) {
 		char buf[PREFIX_STRLEN];
 
 		p = ifc->address;
-		zlog_debug("MESSAGE: ZEBRA_INTERFACE_ADDRESS_ADD %s on %s(%u)",
-			   prefix2str(p, buf, sizeof(buf)), ifp->name,
-			   ifp->vrf_id);
+		snprintf(buf1, sizeof(buf1),
+			 "ZEBRA_INTERFACE_ADDRESS_ADD %s on %s(%u)",
+			 prefix2str(p, buf, sizeof(buf)), ifp->name,
```

Comment:
nit: you could use `snprintfrr` and print the `p` prefix directly using the `%pFX` format specifier (thus avoiding the need for the `buf` local buffer). This applies to other places you're using `prefix2str()` as well.